src/Hazelcast.Net.Tests/Serialization/Compact/ReflectionSerializerTests.cs

```diff
@@ -81,6 +81,40 @@ public void SchemaBuilderWriterExceptions()
 
         private static (Type, object?)[] SerializeSource => ReflectionDataSource.TypeValueList;
 
+        private static readonly (Type, object?)[] SerializeExceptionSource =
+        {
+            (typeof (List<List<int>>), null),
+            (typeof (List<List<int>>), new List<List<int>>()),
+        };
+
+        private static HashSet<int> GetHashSet()
+        {
+            var set = new HashSet<int>();
+            for (var i = 0; i < 10; i++) set.Add(i);
+            return set;
+        }
+
+        private static Dictionary<int, string> GetDictionary()
+        {
+            var dict = new Dictionary<int, string>();
+            for (var i = 0; i < 10; i++) dict[i] = i.ToString();
+            return dict;
+        }
+
+        [TestCaseSource(nameof(SerializeExceptionSource))]
+        public void SerializeOneExceptions((Type PropertyType, object? PropertyValue) testCase)
```

Comment:
What do you test here? Do you test `ReflectionSerializer` throws if the property defined/added to type at runtime?

Reply:
testing that some invalid values for collections actually trigger an exception